src/python/pants/backend/go/target_type_rules.py

```diff
@@ -206,7 +207,7 @@ async def generate_targets_from_go_mod(
     union_membership: UnionMembership,
 ) -> GeneratedTargets:
     generator_addr = request.generator.address
-    go_mod_info = await Get(GoModInfo, GoModInfoRequest(generator_addr))
+    go_mod_info = await Get(GoModInfo, GoModInfoRequest(request.generator[GoModSourcesField]))
```

Comment:
`GoModInfoRequest` was used in target generation, but also requested a `WrappedTarget`... which now requests target generation under the hood. To avoid the cycle, we directly pass the sources to use rather than the `Address` at this callsite.